lms/djangoapps/instructor/tests/views/test_instructor_dashboard.py

```diff
@@ -252,3 +268,51 @@ def test_grade_cutoffs(self):
         """
         response = self.client.get(self.url)
         self.assertIn('D: 0.5, C: 0.57, B: 0.63, A: 0.75', response.content)
+
+    @patch('instructor.views.api.MAX_STUDENTS_PER_PAGE_GRADE_BOOK', 2)
+    def test_calculate_page_info(self):
+        page = calculate_page_info(offset=0, total_students=2)
+        self.assertEqual(page["offset"], 0)
+        self.assertEqual(page["page_num"], 1)
+        self.assertEqual(page["next_offset"], None)
+        self.assertEqual(page["previous_offset"], None)
+        self.assertEqual(page["total_pages"], 1)
+
+    @patch('instructor.views.api.render_to_response', intercept_renderer)
+    @patch('instructor.views.api.MAX_STUDENTS_PER_PAGE_GRADE_BOOK', 1)
+    def test_spoc_gradebook_pages(self):
+        student = UserFactory.create()
+        CourseEnrollmentFactory.create(user=student, course_id=self.course.id)
+
+        student2 = UserFactory.create()
+        CourseEnrollmentFactory.create(user=student2, course_id=self.course.id)
+
+        # create grades for self.student as if they'd submitted the ccx
+        for chapter in self.course.get_children():
+            for i, section in enumerate(chapter.get_children()):
+                for j, problem in enumerate(section.get_children()):
+                    # if not problem.visible_to_staff_only:
+                    StudentModuleFactory.create(
+                        grade=1 if i < j else 0,
+                        max_grade=1,
+                        student=student,
+                        course_id=self.course.id,
+                        module_state_key=problem.location
+                    )
+
+                    StudentModuleFactory.create(
+                        grade=1 if i > j else 0,
+                        max_grade=1,
+                        student=student2,
+                        course_id=self.course.id,
+                        module_state_key=problem.location
+                    )
+
+        url = reverse(
+            'spoc_gradebook',
+            kwargs={'course_id': self.course.id}
+        )
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        # Max number of student per page is one.  Patched setting MAX_STUDENTS_PER_PAGE_GRADE_BOOK = 1
+        self.assertEqual(len(response.mako_context['students']), 1)  # pylint: disable=no-member
```

Comment:
Still getting issue on jenkins, so adding it back.

318: E1101: (no-member), TestInstructorDashboard.test_spoc_gradebook_pages: Instance of 'WSGIRequest' has no 'mako_context' member